### PR TITLE
[FEAT] payprov2

### PR DESCRIPTION
**Pull request by gabrielbazan7**

### CODE REVIEW

File: src/providers/wallet/wallet.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ export interface WalletOptions {
 
 export interface TransactionProposal {
   amount: any;
+  data: string;
```
Comment (justinkook) [inline src/providers/wallet/wallet.ts]: optional `data?:`

File: src/pages/send/confirm/confirm.ts [hunk 2]
```diff
@@ -437,7 +438,8 @@ export class ConfirmPage extends WalletTabsChild {
 
       const maxAllowedMerchantFee = {
         btc: 'urgent',
-        bch: 'normal'
+        bch: 'normal',
+        eth: 'normal'
```
Comment (justinkook) [inline src/pages/send/confirm/confirm.ts]: change to `eth: 'urgent'`

File: src/providers/paypro/paypro.ts [hunk 3]
```diff
@@ -1,104 +1,39 @@
...
-              resolve(paypro);
-            }
-          }
-        );
-      });
+    const bwc = this.bwcProvider.getPayProV2();
+    const options: any = {
+      paymentUrl,
+      chain: coin.toUpperCase(),
+      currency: ''
```
Comment (justinkook) [inline src/providers/paypro/paypro.ts]: use coin.toUpperCase() for currency.

File: src/pages/send/send.ts [hunk 4]
```diff
@@ -129,6 +129,7 @@ export class SendPage extends WalletTabsChild {
         this.wallet.network,
         data
       );
+      isValid = true; // TODO REMOVE THIS
```
Comment (cmgustavo) [inline src/pages/send/send.ts]: +1

File: src/providers/incoming-data/incoming-data.ts [hunk 5]
```diff
@@ -802,11 +821,29 @@ export class IncomingDataProvider {
...
+    if (paymentUrl.indexOf('bitcoin') === 0) {
+      coin = Coin.BTC;
+      paymentUrl = decodeURIComponent(
+        paymentUrl.replace(/bitcoin(cash)?:\?r=/, '')
+      );
+    }
+    if (paymentUrl.indexOf('ethereum') === 0) {
+      coin = Coin.ETH;
+      paymentUrl = decodeURIComponent(paymentUrl.replace(/ethereum?:\?r=/, ''));
+    }
```
Comment (justinkook) [inline src/providers/incoming-data/incoming-data.ts]: Use `getCoinFromUri` and `getPayProUrl` functions here to not have any merge conflicts into eth branch.

https://github.com/bitpay/copay/pull/10179/files#diff-879a6f0c0c6622976ede2f66f41ab9f6R730-R755

**Pull request merged**